Services/ConsentService.cs

```diff
@@ -0,0 +1,98 @@
+using Lombiq.Privacy.Models;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using OrchardCore.Entities;
+using OrchardCore.Users;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Lombiq.Privacy.Services
+{
+    public class ConsentService : IConsentService
+    {
+        private readonly UserManager<IUser> _userManager;
+        private readonly IUserService _userService;
+
+        private readonly IOptions<CookiePolicyOptions> _cookiePolicyOptions;
+
+        public ConsentService(UserManager<IUser> userManager, IOptions<CookiePolicyOptions> cookiePolicyOptions, IUserService userService)
+        {
+            _userManager = userManager;
+            _cookiePolicyOptions = cookiePolicyOptions;
+            _userService = userService;
+        }
+
+        public async Task<bool> IsConsentBannerNeededAsync(HttpContext httpContext)
+        {
+            var consentFeature = httpContext.Features.Get<ITrackingConsentFeature>();
+            if (!consentFeature.IsConsentNeeded)
+            {
+                return false;
+            }
+
+            if (httpContext.User.Identity.IsAuthenticated)
+            {
+                var user = await _userService.GetAuthenticatedUserAsync(httpContext.User);
+
+                return user is not User orchardUser || !orchardUser.Has<PrivacyConsent>();
```

Comment:
Ha `orchardUser.Has<PrivacyConsent>()`, akkor biztos, hogy `PrivacyConsent.Accepted == true`?

Reply:
Alapvetően ha a `orchardUser.Has<PrivacyConsent>()` az nem jelenti azt hogy el is fogadta. Ez a függvény ugye azt nem azt dönti el hogy elfogadta-e hanem azt hogy egyáltalán mondott e rá bármit(true/false) mert ha nem akkor itt az ideje :)

Máskérdés hogy most jelenleg úgy van megcsinálva hogy autentikált user csak elfogadni tudja. Ezt változtassam meg úgy hogy elutasítani is tudja?(db-be false tárolódjon le)?

Reply:
Ezt azért így irtam mert ha lehetőség lenne regisztrált usernek nem elfogadni a hozzájárulást akkor(`PrivacyConsent.Accepted == false`): 
`IsConsentNeededAsync()`=>`true` kell hogy viszaadjon azért hogy a háttérben a CanTrack false legyen. (jelen implementációban telejesül)
`IsConsentBannerNeededAsync()` => `false` kell hogy adjon mert már válaszolt a "kérdésre" és nem vlot a válasz(azaz van PrivacyConsent-je. Jelen implementációban ez is teljesül)

Reply:
Aha, OK akkor.